### PR TITLE
adds snippet type to the url for copy url

### PR DESCRIPTION
**Pull request by mrudulpatil18**
Issue: #129

- Updated encodeTextToUrlHash to add snippet type to the URL.
- If not explicitly specified : Defaults to board if <board is detected in code; otherwise package.

### CODE REVIEW

File: src/lib/encodeTextToUrlHash.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
 import { gzipSync, strToU8 } from "fflate"
 import { bytesToBase64 } from "./bytesToBase64"
 
-export function encodeTextToUrlHash(text: string): string {
+export function encodeTextToUrlHash(text: string, type?: string): string {
```
Comment (seveibar) [inline src/lib/encodeTextToUrlHash.ts:4]:
```suggestion
export function encodeTextToUrlHash(text: string, snippet_type?: string): string {
```

File: src/lib/encodeTextToUrlHash.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
   // Compress the input string
   const compressedData = gzipSync(strToU8(text))
 
   // Convert to base64
   const base64Data = bytesToBase64(compressedData)
 
   // Construct the URL
-  return `${window.location.origin}/editor#data:application/gzip;base64,${base64Data}`
+  const typeParam = type ? `&type=${type}` : ""
```
Comment (seveibar) [inline src/lib/encodeTextToUrlHash.ts:12]:
```suggestion
  const typeParam = snippet_type ? `&snippet_type=${type}` : ""
```

**Review by seveibar — APPROVED** [review]
always better to keep prefixes with `type`, `data` or other vars that are often ambiguous in different contexts. e.g. we could support `order_type`, `template_type` or other url params in the future with that kind of name

File: src/lib/encodeTextToUrlHash.ts (hunk 2 above)
Comment (mrudulpatil18) [reply src/lib/encodeTextToUrlHash.ts]: modified in recent commit

File: src/lib/encodeTextToUrlHash.ts (hunk 1 above)
Comment (mrudulpatil18) [reply src/lib/encodeTextToUrlHash.ts]: modified in recent commit

**Pull request merged**